cmd/virt-dra-controller/pci.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"fmt"
+
+	pcicrd "kubevirt.io/kubevirt-dra-driver/api/kubevirt.io/resource/pci/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	resourcev1 "k8s.io/api/resource/v1alpha2"
+	"k8s.io/dynamic-resource-allocation/controller"
+
+	nascrd "kubevirt.io/kubevirt-dra-driver/api/kubevirt.io/resource/pci/nas/v1alpha1"
+)
+
+type pcidriver struct {
+	PendingAllocatedClaims *PerNodeAllocatedClaims
+}
+
+func NewPciDriver() *pcidriver {
+	return &pcidriver{
+		PendingAllocatedClaims: NewPerNodeAllocatedClaims(),
+	}
+}
+
+func (p *pcidriver) ValidateClaimParameters(claimParams *pcicrd.PciClaimParametersSpec) error {
+	if claimParams.DeviceName != "devices.kubevirt.io/nvme" {
```

Comment:
I guess this can be any arbitary string, right?

Reply:
No , `DeviceName` is supposed to be same as one of the supported  `resourceName` in `DeviceClassParameters` . 

I think we need to have somekind of mapping of what devices this driver supports . For now i have kept only `devices.kubevirt.io/nvme`

Currently VMI declaration [1] in kubevirt uses something similar as

```
      hostDevices:  
      - name: nvme  
        deviceName: devices.kubevirt.io/nvme  
```

I also think a `name` field needs to added in the `PciClaimParameters`  , which will be arbitary defined by user

[1] https://kubevirt.io/user-guide/compute/host-devices/#nvme-pci-passthrough